source/extensions/listener_managers/listener_manager/listener_impl.cc

```diff
@@ -403,27 +395,27 @@ ListenerImpl::ListenerImpl(const envoy::config::listener::v3::Listener& config,
     ENVOY_LOG(warn,
               "Listener connection limit runtime key {} is empty. There are currently no "
               "limitations on the number of accepted connections for listener {}.",
-              cx_limit_runtime_key_, config_.name());
+              cx_limit_runtime_key_, config.name());
   }
 
   filter_chain_manager_ = std::make_unique<FilterChainManagerImpl>(
       addresses_, listener_factory_context_->parentFactoryContext(), initManager()),
 
-  buildAccessLog();
+  buildAccessLog(config);
   validateConfig();
 
   // buildUdpListenerFactory() must come before buildListenSocketOptions() because the UDP
   // listener factory can provide additional options.
-  buildUdpListenerFactory(parent_.server_.options().concurrency());
-  buildListenSocketOptions(address_opts_list);
-  createListenerFilterFactories();
-  validateFilterChains();
-  buildFilterChains();
+  buildUdpListenerFactory(config, parent_.server_.options().concurrency());
```

Comment:
mild preference for leaving these function definitions as-is and adding a protected accessor config() which returns info->config()

Reply:
I thought it would be good to delete `config()` accessor long term because storing the proto as-is can be expensive with large listeners or Wasm. It's only needed for config dump which can be optimized to recreate the proto like it's done in some other places.